apps/src/p5lab/spritelab/commands/criterionCommands.js

```diff
@@ -512,6 +586,42 @@ export const commands = {
     return result;
   },
 
+  // Returns true if a key press event was logged this frame.
+  keyPressEventFound() {
+    // Only check for values that are new this frame
+    for (let i = this.previous.eventLogLength; i < this.eventLog.length; i++) {
+      if (
+        this.eventLog[i].includes('whenPress: ') ||
+        this.eventLog[i].includes('whilePress: ')
+      ) {
+        return true;
+      }
+    }
+    return false;
+  },
+
+  // Returns true if the background changed this frame.
+  backgroundChanged() {
+    if (this.previous.background) {
+      return this.getBackground() !== this.previous.background;
+    } else {
+      return false;
+    }
+  },
+
+  // Returns true if the title text changed this frame.
+  titleChanged() {
+    const previousScreenText = this.previous.screenText;
```

Comment:
Here and above, is `this.previous` guaranteed to exist?

Reply:
Yes it is!